vnext/Chakra/ChakraJsiRuntime_core.cpp

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "ChakraJsiRuntime.h"
+#include "UnicodeConversion.h"
+
+#include <cxxreact/MessageQueueThread.h>
+
+// This file contains non-edge-mode (or win32) implementations.
```

Comment:
I'm confused about the separation between Chakra (Edge) and ChakraCore. Chakra can target Edge mode when USE_EDGEMODE_JSRT is defined. We're not targeting Chakra (Legacy) when USE_EDGEMODE_JSRT is undefined. Then there's ChakraCore. I think in the past it's been assumed that if USE_EDGEMODE_JSRT is not defined, this means use ChakraCore. That's a tenous way to do it. We should really have a clear way that we define whether ChakraCore or Chakra is being used.  

The breakpoint of ChakraJsiRuntime_core/edgemode.cpp is confusing. It would be better to break the source so there is a JsiRuntime for Chakra and a JsiRuntime for ChakraCore. As the JSRT api is similar for both but different, there can be an underlying set of code that works across both, and then ChakraCoreJsiRuntime (essentially renamed ChakraJsiRuntimeWithDebugger) defined here would provide for ChakraCore, and ChakraJsiRuntime would target Chakra (Edge).

Reply:
Hi Adam,
Thanks for a very informative comment. I don't know enough context and history to take the right decision here. From looking at the old ChakraExecutor flow, it felt to me that USE_EDGEMODE_JSRT macro definition differentiates between the ChakraCore and UWP mode. If that is not the best option, we should definitely reconsider. 
I've renamed the variables to align with our coding conventions.
Thanks for offering me to go ahead with merging the change. We will work closely to straighten the code.